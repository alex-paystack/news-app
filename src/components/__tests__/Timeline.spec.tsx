import React from "react";
import { render, screen } from "@testing-library/react";
import {
  Timeline,
  TimelineItem,
  TimelineDot,
  TimelineTime,
  TimelineContent,
} from "../timeline/Timeline";

test("renders a timeline correctly", () => {
  render(
    <Timeline data-testid="timeline">
      <TimelineItem>
        <TimelineDot />
        <TimelineTime>11:30</TimelineTime>
        <TimelineContent>
          Get access to over 20+ pages including a dashboard layout, charts,
          kanban board, calendar, and pre-order E-commerce & Marketing pages.
        </TimelineContent>
      </TimelineItem>
    </Timeline>
  ); 

  const timeline = screen.getByTestId("timeline");
  expect(timeline).toMatchSnapshot();
});
