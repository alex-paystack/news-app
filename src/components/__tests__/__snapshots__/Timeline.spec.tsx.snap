// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a timeline correctly 1`] = `
<div
  class="timeline"
  data-testid="timeline"
>
  <ol>
    <li
      class="timeline-item"
    >
      <div
        class="timeline-dot"
      >
        <div />
      </div>
      <div
        class="timeline-time"
      >
        11:30
      </div>
      <div
        class="timeline-content"
      >
        Get access to over 20+ pages including a dashboard layout, charts, kanban board, calendar, and pre-order E-commerce & Marketing pages.
      </div>
    </li>
  </ol>
</div>
`;
